app/src/main/java/org/greenstand/android/TreeTracker/login/LoginView.kt

```diff
@@ -0,0 +1,221 @@
+import androidx.compose.ui.platform.LocalTextInputService
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+import kotlinx.coroutines.launch
+import org.greenstand.android.TreeTracker.R
+import org.greenstand.android.TreeTracker.utilities.Padding
+import org.greenstand.android.TreeTracker.utilities.isValidEmail
+import org.greenstand.android.TreeTracker.utilities.isValidPhoneNumber
+import org.greenstand.android.TreeTracker.view.TreeTrackerButton
+import org.greenstand.android.TreeTracker.view.TreeTrackerTheme
+
+@Composable
+fun LoginView(
+    viewModel: LoginViewModel,
+    navController: NavController
+) {
+
+    val uiScope = rememberCoroutineScope()
+    val snackBarHostState = remember { SnackbarHostState() }
+    val phoneTextInputState = mutableStateOf("")
+    val emailTextInputState = mutableStateOf("")
+    val emailFocusRequester = remember { FocusRequester() }
+    val context = LocalContext.current
+    val textInputService = LocalTextInputService.current
+
+    viewModel.uiEvents.observe(LocalLifecycleOwner.current) { uiEvent ->
+        when (uiEvent) {
+            is LoginViewModel.UIEvent.NavigationRequestEvent -> navController.navigate(uiEvent.newNavDirection)
+            is LoginViewModel.UIEvent.ErrorEvent -> uiScope.launch {
+                snackBarHostState.showSnackbar(message = context.getString(uiEvent.errorMessage))
+            }
+            else -> Unit // **Note: Some UIEvents are for the LoginFragment to handle and should be ignored here
+        }
+    }
+
+    TreeTrackerTheme {
+        Scaffold(
+            snackbarHost = {
+                SnackbarHost(hostState = snackBarHostState) { data ->
+                    Snackbar(snackbarData = data)
+                }
+            }
+        ) {
+            Column(
+                modifier = Modifier
+                    .padding(top = 16.dp, start = 16.dp, end = 16.dp, bottom = 19.dp)
+                    .fillMaxSize()
+            ) {
+
+                Image(
+                    modifier = Modifier
+                        .padding(top = 28.dp)
+                        .width(dimensionResource(id = R.dimen.logo_width))
+                        .height(dimensionResource(id = R.dimen.logo_height)),
+
+                    imageVector = ImageVector.vectorResource(id = R.drawable.logo),
+                    contentDescription = stringResource(id = R.string.greenstand_welcome_text)
+                )
+
+                PhoneInputText(
+                    textInputState = phoneTextInputState,
+                    onActionDone = {
+                        emailFocusRequester.requestFocus()
```

Comment:
This method of controlling focus reminds me a lot of Flutter.

Reply:
Yeah I'm not really a fan of it. In all honesty, I just copied the way we handled keyboard actions to move users to another view from my company. For giggles, I'll see if I can find a more intuitive way of doing it.